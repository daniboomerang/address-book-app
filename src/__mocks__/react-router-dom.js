import React from 'react';

const rrd = require('react-router-dom');

// Just render a fragment with its children
rrd.BrowserRouter = ({ children }) => <>{children}</>;
module.exports = rrd;
