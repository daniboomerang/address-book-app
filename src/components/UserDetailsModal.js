import React from 'react';
import ReactModal from 'react-modal';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faPhone, faMobileAlt, faTimes } from '@fortawesome/free-solid-svg-icons';

/**
 * @param {Props} props
 * @param {Object} props.user - User details object
 * @param {boolean} props.isOpen -  Whether the modal is opened or not
 * @param {function} props.onclose - Callback to execute when clicking the close icon
 */
const UserDetailsModal = ({ user, isOpen, onClose }) => {
  const {
    name: { first: firstName, last: lastName },
    login: { username },
    email,
    picture: { large },
    location: {
      street: { name: streetName, number: streetNumber },
      city,
      state,
      postcode,
    },
    phone,
    cell,
  } = user;

  return (
    <ReactModal
      isOpen={isOpen}
      contentLabel={`${user.firstName} ${user.lastName} details`}
      onRequestClose={onClose}
      overlayClassName="fixed flex items-center bg-black z-20 inset-0 bg-opacity-75"
      className="absolute bg-primary outline-none inset-x-0 w-10/12 sm:w-3/5 lg:w-2/5 mx-auto rounded-lg"
      ariaHideApp={false}
    >
      <div
        data-testid="user-details"
        className="border-4 border-blue-500 shadow-xl rounded-lg relative"
      >
        <FontAwesomeIcon
          data-testid="close-modal"
          className="absolute right-0 m-4 transform cursor-pointer duration-200 hover:scale-110 hover:text-red-600 absolute"
          icon={faTimes}
          size="2x"
          onClick={onClose}
        />
        <div className="h-64 px-4 py-8 sm:p-10">
          <p className="font-bold text-2xl sm:text-4xl lg:text-4xl">
            <span data-testid={firstName} className="mr-1">
              {firstName}
            </span>
            <span data-testid={lastName} className="ml-1">
              {lastName}
            </span>
          </p>
          <p className="text-red-600 text-base md:text-lg mt-2">Username</p>
          <p data-testid={username} className="my-1 text-base md:text-lg">
            {username}
          </p>
          <p className="text-red-600 text-base md:text-lg mt-2">Email Address</p>
          <p data-testid={email} className="my-1 text-sm md:text-lg mb-4">
            {email}
          </p>
        </div>
        <div className="h-36 px-4 py-8 sm:p-10 bg-primary border-t-4 border-blue-500 rounded-b-lg">
          <div className="relative">
            <div className="right-0 w-24 h-24 sm:w-32 sm:h-32 rounded-full shadow-lg absolute -mt-24 sm:-mt-32">
              <div className="flex border-4 rounded-full justify-center duration-200 transform w-24 h-24 sm:w-32 sm:h-32 border-red-600">
                <img
                  data-testid={large}
                  className="border-primary rounded-full transform duration-200 border-4"
                  src={large}
                  alt={username}
                />
              </div>
            </div>
          </div>
          <div className="text-blue-500 mb-6">
            <p className="text-red-600 text-base md:text-lg">Address</p>
            <div data-testid="address" className="my-1 text-base md:text-lg mb-4">
              <p>{`${streetNumber} ${streetName},`}</p>
              <p>{`PC: ${postcode}, ${city} ${state}`}</p>
            </div>
          </div>
          <div data-testid="phone-number" className="border-t-2 border-blue-500 flex items-center">
            <FontAwesomeIcon className="m-4 text-red-600" icon={faPhone} size="1x" />
            {phone}
          </div>
          <div data-testid="cell-number" className="border-t-2 border-blue-500 flex items-center">
            <FontAwesomeIcon className="m-4 text-red-600" icon={faMobileAlt} size="1x" />
            <span className="ml-3">{cell}</span>
          </div>
        </div>
      </div>
    </ReactModal>
  );
};

export default UserDetailsModal;
