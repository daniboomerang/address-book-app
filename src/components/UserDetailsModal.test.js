import React from 'react';
import { mount } from 'enzyme';
import { usersMockedData } from '../__mocks__/mocked-data';
import UserDetailsModal from './UserDetailsModal';

describe('UserDetailsModal component', () => {
  const user = usersMockedData[0];
  const {
    name: { first: firstName, last: lastName },
    login: { username },
    email,
    picture: { large },
    location: {
      street: { name: streetName, number: streetNumber },
      city,
      state,
      postcode,
    },
    phone,
    cell,
  } = user;

  const onClose = jest.fn();

  it('should render correctly when the modal is open', () => {
    const wrapper = mount(<UserDetailsModal isOpen user={user} onClose={onClose} />);

    expect(wrapper.find(`div[data-testid="user-details"]`).length).toBe(1);
    expect(wrapper.find(`span[data-testid="${firstName}"]`).text()).toBe(firstName);
    expect(wrapper.find(`span[data-testid="${lastName}"]`).text()).toBe(lastName);
    expect(wrapper.find(`p[data-testid="${username}"]`).text()).toBe(username);
    expect(wrapper.find(`p[data-testid="${email}"]`).text()).toBe(email);
    expect(wrapper.find(`img[src="${large}"]`).length).toBe(1);
    expect(wrapper.find(`div[data-testid="address"]`).text()).toBe(
      `${streetNumber} ${streetName},PC: ${postcode}, ${city} ${state}`
    );
    expect(wrapper.find(`div[data-testid="phone-number"]`).text()).toBe(phone);
    expect(wrapper.find(`div[data-testid="cell-number"]`).text()).toBe(cell);

    expect(wrapper).toMatchSnapshot();
  });

  it('should render correctly when the modal is closed', () => {
    const wrapper = mount(<UserDetailsModal isOpen={false} user={user} onClose={onClose} />);

    expect(wrapper.find(`div[data-testid="user-details"]`).length).toBe(0);

    expect(wrapper).toMatchSnapshot();
  });

  it('should call on close when the X is clicked', () => {
    const wrapper = mount(<UserDetailsModal isOpen user={user} onClose={onClose} />);

    const theX = wrapper.find(`FontAwesomeIcon[data-testid="close-modal"]`);
    expect(theX.length).toBe(1);
    theX.simulate('click');

    expect(onClose).toHaveBeenCalled();
  });
});
