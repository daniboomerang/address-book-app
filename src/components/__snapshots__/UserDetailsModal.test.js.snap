// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserDetailsModal component should render correctly when the modal is closed 1`] = `
<UserDetailsModal
  isOpen={false}
  onClose={[MockFunction]}
  user={
    Object {
      "cell": "88592966",
      "dob": Object {
        "age": 70,
        "date": "1950-02-27T16:49:15.705Z",
      },
      "email": "simmon.martin@example.com",
      "gender": "male",
      "id": Object {
        "name": "CPR",
        "value": "270250-8357",
      },
      "location": Object {
        "city": "Tisvilde",
        "coordinates": Object {
          "latitude": "-27.4297",
          "longitude": "-37.0144",
        },
        "country": "Denmark",
        "postcode": 28816,
        "state": "Danmark",
        "street": Object {
          "name": "Gormsvej",
          "number": 3865,
        },
        "timezone": Object {
          "description": "Abu Dhabi, Muscat, Baku, Tbilisi",
          "offset": "+4:00",
        },
      },
      "login": Object {
        "md5": "ac3673267c4606e6fa5f754047d57a4a",
        "password": "testibil",
        "salt": "uPCA1DOi",
        "sha1": "c1e2a5b2e2daa5020a02b4ba54bc8881b4233b6c",
        "sha256": "4797c6d5a893a92bccca7b09cb8c19381d909389bbfeab3a6a9ec07bc8503b23",
        "username": "purplebird556",
        "uuid": "dca63e13-5a76-4d7a-aea1-817919275f83",
      },
      "name": Object {
        "first": "Simmon",
        "last": "Martin",
        "title": "Mr",
      },
      "nat": "DK",
      "phone": "05143951",
      "picture": Object {
        "large": "https://randomuser.me/api/portraits/men/59.jpg",
        "medium": "https://randomuser.me/api/portraits/med/men/59.jpg",
        "thumbnail": "https://randomuser.me/api/portraits/thumb/men/59.jpg",
      },
      "registered": Object {
        "age": 10,
        "date": "2010-10-27T11:48:05.160Z",
      },
    }
  }
>
  <Modal
    ariaHideApp={false}
    bodyOpenClassName="ReactModal__Body--open"
    className="absolute bg-primary outline-none inset-x-0 w-10/12 sm:w-3/5 lg:w-2/5 mx-auto rounded-lg"
    closeTimeoutMS={0}
    contentLabel="undefined undefined details"
    isOpen={false}
    onRequestClose={[MockFunction]}
    overlayClassName="fixed flex items-center bg-black z-20 inset-0 bg-opacity-75"
    parentSelector={[Function]}
    portalClassName="ReactModalPortal"
    role="dialog"
    shouldCloseOnEsc={true}
    shouldCloseOnOverlayClick={true}
    shouldFocusAfterRender={true}
    shouldReturnFocusAfterClose={true}
  >
    <Portal
      containerInfo={
        <div
          class="ReactModalPortal"
        />
      }
    >
      <ModalPortal
        ariaHideApp={false}
        bodyOpenClassName="ReactModal__Body--open"
        className="absolute bg-primary outline-none inset-x-0 w-10/12 sm:w-3/5 lg:w-2/5 mx-auto rounded-lg"
        closeTimeoutMS={0}
        contentLabel="undefined undefined details"
        defaultStyles={
          Object {
            "content": Object {
              "WebkitOverflowScrolling": "touch",
              "background": "#fff",
              "border": "1px solid #ccc",
              "borderRadius": "4px",
              "bottom": "40px",
              "left": "40px",
              "outline": "none",
              "overflow": "auto",
              "padding": "20px",
              "position": "absolute",
              "right": "40px",
              "top": "40px",
            },
            "overlay": Object {
              "backgroundColor": "rgba(255, 255, 255, 0.75)",
              "bottom": 0,
              "left": 0,
              "position": "fixed",
              "right": 0,
              "top": 0,
            },
          }
        }
        isOpen={false}
        onRequestClose={[MockFunction]}
        overlayClassName="fixed flex items-center bg-black z-20 inset-0 bg-opacity-75"
        parentSelector={[Function]}
        portalClassName="ReactModalPortal"
        role="dialog"
        shouldCloseOnEsc={true}
        shouldCloseOnOverlayClick={true}
        shouldFocusAfterRender={true}
        shouldReturnFocusAfterClose={true}
        style={
          Object {
            "content": Object {},
            "overlay": Object {},
          }
        }
      />
    </Portal>
  </Modal>
</UserDetailsModal>
`;

exports[`UserDetailsModal component should render correctly when the modal is open 1`] = `
<UserDetailsModal
  isOpen={true}
  onClose={[MockFunction]}
  user={
    Object {
      "cell": "88592966",
      "dob": Object {
        "age": 70,
        "date": "1950-02-27T16:49:15.705Z",
      },
      "email": "simmon.martin@example.com",
      "gender": "male",
      "id": Object {
        "name": "CPR",
        "value": "270250-8357",
      },
      "location": Object {
        "city": "Tisvilde",
        "coordinates": Object {
          "latitude": "-27.4297",
          "longitude": "-37.0144",
        },
        "country": "Denmark",
        "postcode": 28816,
        "state": "Danmark",
        "street": Object {
          "name": "Gormsvej",
          "number": 3865,
        },
        "timezone": Object {
          "description": "Abu Dhabi, Muscat, Baku, Tbilisi",
          "offset": "+4:00",
        },
      },
      "login": Object {
        "md5": "ac3673267c4606e6fa5f754047d57a4a",
        "password": "testibil",
        "salt": "uPCA1DOi",
        "sha1": "c1e2a5b2e2daa5020a02b4ba54bc8881b4233b6c",
        "sha256": "4797c6d5a893a92bccca7b09cb8c19381d909389bbfeab3a6a9ec07bc8503b23",
        "username": "purplebird556",
        "uuid": "dca63e13-5a76-4d7a-aea1-817919275f83",
      },
      "name": Object {
        "first": "Simmon",
        "last": "Martin",
        "title": "Mr",
      },
      "nat": "DK",
      "phone": "05143951",
      "picture": Object {
        "large": "https://randomuser.me/api/portraits/men/59.jpg",
        "medium": "https://randomuser.me/api/portraits/med/men/59.jpg",
        "thumbnail": "https://randomuser.me/api/portraits/thumb/men/59.jpg",
      },
      "registered": Object {
        "age": 10,
        "date": "2010-10-27T11:48:05.160Z",
      },
    }
  }
>
  <Modal
    ariaHideApp={false}
    bodyOpenClassName="ReactModal__Body--open"
    className="absolute bg-primary outline-none inset-x-0 w-10/12 sm:w-3/5 lg:w-2/5 mx-auto rounded-lg"
    closeTimeoutMS={0}
    contentLabel="undefined undefined details"
    isOpen={true}
    onRequestClose={[MockFunction]}
    overlayClassName="fixed flex items-center bg-black z-20 inset-0 bg-opacity-75"
    parentSelector={[Function]}
    portalClassName="ReactModalPortal"
    role="dialog"
    shouldCloseOnEsc={true}
    shouldCloseOnOverlayClick={true}
    shouldFocusAfterRender={true}
    shouldReturnFocusAfterClose={true}
  >
    <Portal
      containerInfo={
        <div
          class="ReactModalPortal"
        >
          <div
            class="ReactModal__Overlay ReactModal__Overlay--after-open fixed flex items-center bg-black z-20 inset-0 bg-opacity-75"
          >
            <div
              aria-label="undefined undefined details"
              class="ReactModal__Content ReactModal__Content--after-open absolute bg-primary outline-none inset-x-0 w-10/12 sm:w-3/5 lg:w-2/5 mx-auto rounded-lg"
              role="dialog"
              tabindex="-1"
            >
              <div
                class="border-4 border-blue-500 shadow-xl rounded-lg relative"
                data-testid="user-details"
              >
                <svg
                  aria-hidden="true"
                  class="svg-inline--fa fa-times fa-w-11 fa-2x absolute right-0 m-4 transform cursor-pointer duration-200 hover:scale-110 hover:text-red-600 absolute"
                  data-icon="times"
                  data-prefix="fas"
                  data-testid="close-modal"
                  focusable="false"
                  role="img"
                  viewBox="0 0 352 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
                    fill="currentColor"
                  />
                </svg>
                <div
                  class="h-64 px-4 py-8 sm:p-10"
                >
                  <p
                    class="font-bold text-2xl sm:text-4xl lg:text-4xl"
                  >
                    <span
                      class="mr-1"
                      data-testid="Simmon"
                    >
                      Simmon
                    </span>
                    <span
                      class="ml-1"
                      data-testid="Martin"
                    >
                      Martin
                    </span>
                  </p>
                  <p
                    class="text-red-600 text-base md:text-lg mt-2"
                  >
                    Username
                  </p>
                  <p
                    class="my-1 text-base md:text-lg"
                    data-testid="purplebird556"
                  >
                    purplebird556
                  </p>
                  <p
                    class="text-red-600 text-base md:text-lg mt-2"
                  >
                    Email Address
                  </p>
                  <p
                    class="my-1 text-sm md:text-lg mb-4"
                    data-testid="simmon.martin@example.com"
                  >
                    simmon.martin@example.com
                  </p>
                </div>
                <div
                  class="h-36 px-4 py-8 sm:p-10 bg-primary border-t-4 border-blue-500 rounded-b-lg"
                >
                  <div
                    class="relative"
                  >
                    <div
                      class="right-0 w-24 h-24 sm:w-32 sm:h-32 rounded-full shadow-lg absolute -mt-24 sm:-mt-32"
                    >
                      <div
                        class="flex border-4 rounded-full justify-center duration-200 transform w-24 h-24 sm:w-32 sm:h-32 border-red-600"
                      >
                        <img
                          alt="purplebird556"
                          class="border-primary rounded-full transform duration-200 border-4"
                          data-testid="https://randomuser.me/api/portraits/men/59.jpg"
                          src="https://randomuser.me/api/portraits/men/59.jpg"
                        />
                      </div>
                    </div>
                  </div>
                  <div
                    class="text-blue-500 mb-6"
                  >
                    <p
                      class="text-red-600 text-base md:text-lg"
                    >
                      Address
                    </p>
                    <div
                      class="my-1 text-base md:text-lg mb-4"
                      data-testid="address"
                    >
                      <p>
                        3865 Gormsvej,
                      </p>
                      <p>
                        PC: 28816, Tisvilde Danmark
                      </p>
                    </div>
                  </div>
                  <div
                    class="border-t-2 border-blue-500 flex items-center"
                    data-testid="phone-number"
                  >
                    <svg
                      aria-hidden="true"
                      class="svg-inline--fa fa-phone fa-w-16 fa-1x m-4 text-red-600"
                      data-icon="phone"
                      data-prefix="fas"
                      focusable="false"
                      role="img"
                      viewBox="0 0 512 512"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M493.4 24.6l-104-24c-11.3-2.6-22.9 3.3-27.5 13.9l-48 112c-4.2 9.8-1.4 21.3 6.9 28l60.6 49.6c-36 76.7-98.9 140.5-177.2 177.2l-49.6-60.6c-6.8-8.3-18.2-11.1-28-6.9l-112 48C3.9 366.5-2 378.1.6 389.4l24 104C27.1 504.2 36.7 512 48 512c256.1 0 464-207.5 464-464 0-11.2-7.7-20.9-18.6-23.4z"
                        fill="currentColor"
                      />
                    </svg>
                    05143951
                  </div>
                  <div
                    class="border-t-2 border-blue-500 flex items-center"
                    data-testid="cell-number"
                  >
                    <svg
                      aria-hidden="true"
                      class="svg-inline--fa fa-mobile-alt fa-w-10 fa-1x m-4 text-red-600"
                      data-icon="mobile-alt"
                      data-prefix="fas"
                      focusable="false"
                      role="img"
                      viewBox="0 0 320 512"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M272 0H48C21.5 0 0 21.5 0 48v416c0 26.5 21.5 48 48 48h224c26.5 0 48-21.5 48-48V48c0-26.5-21.5-48-48-48zM160 480c-17.7 0-32-14.3-32-32s14.3-32 32-32 32 14.3 32 32-14.3 32-32 32zm112-108c0 6.6-5.4 12-12 12H60c-6.6 0-12-5.4-12-12V60c0-6.6 5.4-12 12-12h200c6.6 0 12 5.4 12 12v312z"
                        fill="currentColor"
                      />
                    </svg>
                    <span
                      class="ml-3"
                    >
                      88592966
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      }
    >
      <ModalPortal
        ariaHideApp={false}
        bodyOpenClassName="ReactModal__Body--open"
        className="absolute bg-primary outline-none inset-x-0 w-10/12 sm:w-3/5 lg:w-2/5 mx-auto rounded-lg"
        closeTimeoutMS={0}
        contentLabel="undefined undefined details"
        defaultStyles={
          Object {
            "content": Object {
              "WebkitOverflowScrolling": "touch",
              "background": "#fff",
              "border": "1px solid #ccc",
              "borderRadius": "4px",
              "bottom": "40px",
              "left": "40px",
              "outline": "none",
              "overflow": "auto",
              "padding": "20px",
              "position": "absolute",
              "right": "40px",
              "top": "40px",
            },
            "overlay": Object {
              "backgroundColor": "rgba(255, 255, 255, 0.75)",
              "bottom": 0,
              "left": 0,
              "position": "fixed",
              "right": 0,
              "top": 0,
            },
          }
        }
        isOpen={true}
        onRequestClose={[MockFunction]}
        overlayClassName="fixed flex items-center bg-black z-20 inset-0 bg-opacity-75"
        parentSelector={[Function]}
        portalClassName="ReactModalPortal"
        role="dialog"
        shouldCloseOnEsc={true}
        shouldCloseOnOverlayClick={true}
        shouldFocusAfterRender={true}
        shouldReturnFocusAfterClose={true}
        style={
          Object {
            "content": Object {},
            "overlay": Object {},
          }
        }
      >
        <div
          className="ReactModal__Overlay ReactModal__Overlay--after-open fixed flex items-center bg-black z-20 inset-0 bg-opacity-75"
          onClick={[Function]}
          onMouseDown={[Function]}
          style={Object {}}
        >
          <div
            aria-label="undefined undefined details"
            className="ReactModal__Content ReactModal__Content--after-open absolute bg-primary outline-none inset-x-0 w-10/12 sm:w-3/5 lg:w-2/5 mx-auto rounded-lg"
            onClick={[Function]}
            onKeyDown={[Function]}
            onMouseDown={[Function]}
            onMouseUp={[Function]}
            role="dialog"
            style={Object {}}
            tabIndex="-1"
          >
            <div
              className="border-4 border-blue-500 shadow-xl rounded-lg relative"
              data-testid="user-details"
            >
              <FontAwesomeIcon
                border={false}
                className="absolute right-0 m-4 transform cursor-pointer duration-200 hover:scale-110 hover:text-red-600 absolute"
                data-testid="close-modal"
                fixedWidth={false}
                flip={null}
                icon={
                  Object {
                    "icon": Array [
                      352,
                      512,
                      Array [],
                      "f00d",
                      "M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z",
                    ],
                    "iconName": "times",
                    "prefix": "fas",
                  }
                }
                inverse={false}
                listItem={false}
                mask={null}
                onClick={[MockFunction]}
                pull={null}
                pulse={false}
                rotation={null}
                size="2x"
                spin={false}
                swapOpacity={false}
                symbol={false}
                title=""
                transform={null}
              >
                <svg
                  aria-hidden="true"
                  className="svg-inline--fa fa-times fa-w-11 fa-2x absolute right-0 m-4 transform cursor-pointer duration-200 hover:scale-110 hover:text-red-600 absolute"
                  data-icon="times"
                  data-prefix="fas"
                  data-testid="close-modal"
                  focusable="false"
                  onClick={[MockFunction]}
                  role="img"
                  style={Object {}}
                  viewBox="0 0 352 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
                    fill="currentColor"
                    style={Object {}}
                  />
                </svg>
              </FontAwesomeIcon>
              <div
                className="h-64 px-4 py-8 sm:p-10"
              >
                <p
                  className="font-bold text-2xl sm:text-4xl lg:text-4xl"
                >
                  <span
                    className="mr-1"
                    data-testid="Simmon"
                  >
                    Simmon
                  </span>
                  <span
                    className="ml-1"
                    data-testid="Martin"
                  >
                    Martin
                  </span>
                </p>
                <p
                  className="text-red-600 text-base md:text-lg mt-2"
                >
                  Username
                </p>
                <p
                  className="my-1 text-base md:text-lg"
                  data-testid="purplebird556"
                >
                  purplebird556
                </p>
                <p
                  className="text-red-600 text-base md:text-lg mt-2"
                >
                  Email Address
                </p>
                <p
                  className="my-1 text-sm md:text-lg mb-4"
                  data-testid="simmon.martin@example.com"
                >
                  simmon.martin@example.com
                </p>
              </div>
              <div
                className="h-36 px-4 py-8 sm:p-10 bg-primary border-t-4 border-blue-500 rounded-b-lg"
              >
                <div
                  className="relative"
                >
                  <div
                    className="right-0 w-24 h-24 sm:w-32 sm:h-32 rounded-full shadow-lg absolute -mt-24 sm:-mt-32"
                  >
                    <div
                      className="flex border-4 rounded-full justify-center duration-200 transform w-24 h-24 sm:w-32 sm:h-32 border-red-600"
                    >
                      <img
                        alt="purplebird556"
                        className="border-primary rounded-full transform duration-200 border-4"
                        data-testid="https://randomuser.me/api/portraits/men/59.jpg"
                        src="https://randomuser.me/api/portraits/men/59.jpg"
                      />
                    </div>
                  </div>
                </div>
                <div
                  className="text-blue-500 mb-6"
                >
                  <p
                    className="text-red-600 text-base md:text-lg"
                  >
                    Address
                  </p>
                  <div
                    className="my-1 text-base md:text-lg mb-4"
                    data-testid="address"
                  >
                    <p>
                      3865 Gormsvej,
                    </p>
                    <p>
                      PC: 28816, Tisvilde Danmark
                    </p>
                  </div>
                </div>
                <div
                  className="border-t-2 border-blue-500 flex items-center"
                  data-testid="phone-number"
                >
                  <FontAwesomeIcon
                    border={false}
                    className="m-4 text-red-600"
                    fixedWidth={false}
                    flip={null}
                    icon={
                      Object {
                        "icon": Array [
                          512,
                          512,
                          Array [],
                          "f095",
                          "M493.4 24.6l-104-24c-11.3-2.6-22.9 3.3-27.5 13.9l-48 112c-4.2 9.8-1.4 21.3 6.9 28l60.6 49.6c-36 76.7-98.9 140.5-177.2 177.2l-49.6-60.6c-6.8-8.3-18.2-11.1-28-6.9l-112 48C3.9 366.5-2 378.1.6 389.4l24 104C27.1 504.2 36.7 512 48 512c256.1 0 464-207.5 464-464 0-11.2-7.7-20.9-18.6-23.4z",
                        ],
                        "iconName": "phone",
                        "prefix": "fas",
                      }
                    }
                    inverse={false}
                    listItem={false}
                    mask={null}
                    pull={null}
                    pulse={false}
                    rotation={null}
                    size="1x"
                    spin={false}
                    swapOpacity={false}
                    symbol={false}
                    title=""
                    transform={null}
                  >
                    <svg
                      aria-hidden="true"
                      className="svg-inline--fa fa-phone fa-w-16 fa-1x m-4 text-red-600"
                      data-icon="phone"
                      data-prefix="fas"
                      focusable="false"
                      role="img"
                      style={Object {}}
                      viewBox="0 0 512 512"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M493.4 24.6l-104-24c-11.3-2.6-22.9 3.3-27.5 13.9l-48 112c-4.2 9.8-1.4 21.3 6.9 28l60.6 49.6c-36 76.7-98.9 140.5-177.2 177.2l-49.6-60.6c-6.8-8.3-18.2-11.1-28-6.9l-112 48C3.9 366.5-2 378.1.6 389.4l24 104C27.1 504.2 36.7 512 48 512c256.1 0 464-207.5 464-464 0-11.2-7.7-20.9-18.6-23.4z"
                        fill="currentColor"
                        style={Object {}}
                      />
                    </svg>
                  </FontAwesomeIcon>
                  05143951
                </div>
                <div
                  className="border-t-2 border-blue-500 flex items-center"
                  data-testid="cell-number"
                >
                  <FontAwesomeIcon
                    border={false}
                    className="m-4 text-red-600"
                    fixedWidth={false}
                    flip={null}
                    icon={
                      Object {
                        "icon": Array [
                          320,
                          512,
                          Array [],
                          "f3cd",
                          "M272 0H48C21.5 0 0 21.5 0 48v416c0 26.5 21.5 48 48 48h224c26.5 0 48-21.5 48-48V48c0-26.5-21.5-48-48-48zM160 480c-17.7 0-32-14.3-32-32s14.3-32 32-32 32 14.3 32 32-14.3 32-32 32zm112-108c0 6.6-5.4 12-12 12H60c-6.6 0-12-5.4-12-12V60c0-6.6 5.4-12 12-12h200c6.6 0 12 5.4 12 12v312z",
                        ],
                        "iconName": "mobile-alt",
                        "prefix": "fas",
                      }
                    }
                    inverse={false}
                    listItem={false}
                    mask={null}
                    pull={null}
                    pulse={false}
                    rotation={null}
                    size="1x"
                    spin={false}
                    swapOpacity={false}
                    symbol={false}
                    title=""
                    transform={null}
                  >
                    <svg
                      aria-hidden="true"
                      className="svg-inline--fa fa-mobile-alt fa-w-10 fa-1x m-4 text-red-600"
                      data-icon="mobile-alt"
                      data-prefix="fas"
                      focusable="false"
                      role="img"
                      style={Object {}}
                      viewBox="0 0 320 512"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M272 0H48C21.5 0 0 21.5 0 48v416c0 26.5 21.5 48 48 48h224c26.5 0 48-21.5 48-48V48c0-26.5-21.5-48-48-48zM160 480c-17.7 0-32-14.3-32-32s14.3-32 32-32 32 14.3 32 32-14.3 32-32 32zm112-108c0 6.6-5.4 12-12 12H60c-6.6 0-12-5.4-12-12V60c0-6.6 5.4-12 12-12h200c6.6 0 12 5.4 12 12v312z"
                        fill="currentColor"
                        style={Object {}}
                      />
                    </svg>
                  </FontAwesomeIcon>
                  <span
                    className="ml-3"
                  >
                    88592966
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
      </ModalPortal>
    </Portal>
  </Modal>
</UserDetailsModal>
`;
